persistence/src/commonTest/kotlin/com/wire/kalium/persistence/dao/SyncDAOTest.kt

```diff
@@ -0,0 +1,62 @@
+/*
+ * Wire
+ * Copyright (C) 2023 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+package com.wire.kalium.persistence.dao
+
+import com.wire.kalium.persistence.BaseDatabaseTest
+import com.wire.kalium.persistence.utils.stubs.newUserEntity
+import kotlinx.coroutines.test.runTest
+import kotlin.test.BeforeTest
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertFalse
+
+class SyncDAOTest: BaseDatabaseTest() {
+    lateinit var userDAO: UserDAO
+    lateinit var syncDAO: SyncDAO
+
+    val SELF_USER_ID = UserIDEntity("self-id", "domain")
+
+    @BeforeTest
+    fun setUp() {
+        deleteDatabase(SELF_USER_ID)
+        val db = createDatabase(SELF_USER_ID, encryptedDBSecret, true)
+        userDAO = db.userDAO
+        syncDAO = db.syncDAO
+    }
+
+    @Test
+    fun givenUsersId_whenCallingAllOtherUsers_thenSelfIdIsNotIncluded() = runTest {
+        val selfUser = newUserEntity().copy(id = SELF_USER_ID)
+        val user1 = newUserEntity().copy(id = UserIDEntity("user-1", "domain-1"))
+        val user2 = newUserEntity().copy(id = UserIDEntity("user-2", "domain-2"))
+        val user3 = newUserEntity().copy(id = UserIDEntity("user-3", "domain-1"))
+
+        userDAO.insertUser(selfUser)
+        userDAO.insertUser(user1)
+        userDAO.insertUser(user2)
+        userDAO.insertUser(user3)
+
+        syncDAO.allOtherUsersId().also { result ->
+            assertFalse {
+                result.contains(selfUser.id)
+            }
+            assertEquals(result, listOf(user1.id, user2.id, user3.id))
+        }
+
```

Comment:
We don't use `syncDAO` to insert stuff. It's just to read and filter from another DAO. That's the part that to me makes it extra-obvious that something is out of place.

- "How can I get all known users, except self user?"
- "Oh, just check SyncDAO"
- "?"

`UserDAO` would make it obvious and easy to discover.

Reply:
UserDAO already have `@Suppress("TooManyFunctions")` and i would like to offload functionality from it not add new one, that is where the SyncDAO naming comes from

Reply:
It's easier to just move some of the functions there to `MemberDAO` instead, like:

`getUsersNotInConversationByHandle`
`getUsersNotInConversationByNameOrHandleOrEmail`
`observeUsersNotInConversation`

`TooManyFunctions` is one indication that a class _might_ have far too many responsibilities. But it is not necessarily a deal breaker. This function suits `UserDAO` and `Sync` is just a weird thing to have in the persistence level.